presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitSource.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.hudi;
+
+import com.facebook.presto.hive.metastore.ExtendedHiveMetastore;
+import com.facebook.presto.hive.util.AsyncQueue;
+import com.facebook.presto.hudi.split.HudiBackgroundSplitLoader;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.ConnectorSplitSource;
+import com.facebook.presto.spi.connector.ConnectorPartitionHandle;
+import com.google.common.util.concurrent.Futures;
+import org.apache.hudi.common.table.view.HoodieTableFileSystemView;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.airlift.concurrent.MoreFutures.toCompletableFuture;
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static java.util.Objects.requireNonNull;
+
+public class HudiSplitSource
+        implements ConnectorSplitSource
+{
+    private final AsyncQueue<ConnectorSplit> queue;
+    private final HudiBackgroundSplitLoader splitLoader;
+    private final ScheduledExecutorService splitLoaderExecutorService;
```

Comment:
better to inline these in the constructor instead of having member variables.

Reply:
Good catch!  Fixed.